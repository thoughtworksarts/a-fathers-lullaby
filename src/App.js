import React from 'react';
import Header from './components/molecules/Header/Header';
import AboutPage from './pages/About/About';
import './App.css';

function App() {
  return (
    <div className='App'>
      <Header />
      <AboutPage />
    </div>
  );
}

export default App;
