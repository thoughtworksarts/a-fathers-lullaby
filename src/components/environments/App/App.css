.App {
    z-index: 0;
}