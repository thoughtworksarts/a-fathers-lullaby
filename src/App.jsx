import React from 'react'
import Menu from './components/Menu/Menu'
import AboutPage from './pages/About/About'
import './App.css'

function App () {
  return (
    <div className='App'>
      <Menu />
      <AboutPage />
    </div>
  )
}

export default App
