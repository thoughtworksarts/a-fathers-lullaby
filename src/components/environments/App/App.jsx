import React, { useEffect } from 'react'
import { Switch, Route } from 'react-router-dom'
import { withTracking } from 'helpers'
import { Header } from 'organisms'
import { About, Share, Listen } from 'pages'
import './App.css'

const App = ({ onPageViewTracking }) => {
  // used to track page views and send to google analytics
  useEffect(() => {
    onPageViewTracking()
  }, [window.location.pathname])

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route exact path='/' component={About} />
        <Route path='/share' component={Share} />
        <Route path='/listen' component={Listen} />
      </Switch>
    </div>
  )
}

export default withTracking(App)
