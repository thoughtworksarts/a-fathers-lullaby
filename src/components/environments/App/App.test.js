/* eslint-env mocha */

import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter as Router } from 'react-router-dom'
import App from './App'
import { shallow } from 'enzyme'

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Router>
        <App />
      </Router>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  it('runs an Enzyme test without crashing', () => {
    const appComponent = shallow(<App />)
    expect(appComponent).toMatchSnapshot()
  })
})
