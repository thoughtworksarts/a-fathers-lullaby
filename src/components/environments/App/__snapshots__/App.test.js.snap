// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App runs an Enzyme test without crashing 1`] = `ShallowWrapper {}`;
